packages/runtime/container-runtime/src/test/gc/garbageCollection.spec.ts

```diff
@@ -696,13 +730,302 @@ describe("Garbage Collection Tests", () => {
 
 			summarizerContainerTests(
 				sweepTimeoutMs,
+				"sweep",
 				"GarbageCollector:SweepReadyObject_Revived",
 				"GarbageCollector:SweepReadyObject_Changed",
 				"GarbageCollector:SweepReadyObject_Loaded",
 				false, // expectDeleteLogs
 			);
 		});
 
+		describe("GC version changes", () => {
+			type GcWithStateOverride = IGarbageCollector & {
+				readonly baseSnapshotDataP: Promise<IGarbageCollectionSnapshotData | undefined>;
+				readonly tombstones: string[];
+				readonly deletedNodes: Set<string>;
+				readonly summaryStateTracker: Omit<GCSummaryStateTracker, "latestSummaryData"> & {
+					latestSummaryData: IGCSummaryTrackingData | undefined;
+				};
+			};
+
+			const createGCOverride = (
+				baseSnapshot: ISnapshotTree,
+				gcBlobsMap: Map<
+					string,
+					IGarbageCollectionState | IGarbageCollectionDetailsBase | string[]
+				>,
+				gcFeature: GCVersion,
+			) => {
+				const gcMetadata: IGCMetadata = {
+					gcFeature,
+				};
+				return createGarbageCollector(
+					{ baseSnapshot },
+					gcBlobsMap,
+					gcMetadata,
+				) as GcWithStateOverride;
+			};
+
+			function getSnapshotWithGCVersion(gcVersion: GCVersion) {
+				// Create a snapshot tree to be used as the GC snapshot tree.
+				const gcSnapshotTree = getDummySnapshotTree();
+				const gcBlobId = "root";
+				// Add a GC blob with key that start with `gcBlobPrefix` to the GC snapshot tree. The blob Id for this
+				// is generated by server in real scenarios but we use a static id here for testing.
+				gcSnapshotTree.blobs[`${gcBlobPrefix}_${gcBlobId}`] = gcBlobId;
+
+				// Create GC state with a node. This will be returned when the garbage collector asks for the GC blob
+				// with `gcBlobId`.
+				const gcState: IGarbageCollectionState = { gcNodes: {} };
+				const nodeData: IGarbageCollectionNodeData = {
+					outboundRoutes: [],
+					unreferencedTimestampMs: 123,
+				};
+				gcState.gcNodes[nodes[0]] = nodeData;
+
+				// Create a tombstone blob a node as tombstone. This will be returned when the garbage collector asks for
+				// the tombstone blob.
+				const gcTombstoneBlobId = "tombstone";
+				gcSnapshotTree.blobs[gcTombstoneBlobKey] = gcTombstoneBlobId;
+				const tombstones = [nodes[0]];
+
+				// Create a tombstone blob a node as tombstone. This will be returned when the garbage collector asks for
+				// the tombstone blob.
+				const gcDeletedBlobId = "deletedNodes";
+				gcSnapshotTree.blobs[gcDeletedBlobKey] = gcDeletedBlobId;
+				const deletedBlobs = [nodes[0]];
```

Comment:
Is it intentional that you're putting the same node in both tombstone and deleted list? I guess that's what will happen for a document that's had tombstone enabled and then switches to Sweep?

Reply:
That wasn't the intention. These tests don't care about the contents of tombstone / deleted nodes. They just care that they exist.